cmd/primarycmd/addValidator.go

```diff
@@ -0,0 +1,202 @@
+	txt := "What is the public key of the node's BLS?"
+	publicKey, err := app.Prompt.CaptureString(txt)
+	if err != nil {
+		return jsonProofOfPossession{}, err
+	}
+	txt = "What is the proof of possession of the node's BLS?"
+	pop, err := app.Prompt.CaptureString(txt)
+	if err != nil {
+		return jsonProofOfPossession{}, err
+	}
+	return jsonProofOfPossession{PublicKey: publicKey, ProofOfPossession: pop}, nil
+}
+
+func addValidator(_ *cobra.Command, _ []string) error {
+	var (
+		nodeID ids.NodeID
+		start  time.Time
+		err    error
+	)
+
+	var network models.Network
+	switch {
+	case deployTestnet:
+		network = models.Fuji
+	case deployMainnet:
+		network = models.Mainnet
+	}
+
+	if network == models.Undefined {
+		networkStr, err := app.Prompt.CaptureList(
+			"Choose a network to add validator to.",
+			[]string{models.Fuji.String(), models.Mainnet.String()},
+		)
+		if err != nil {
+			return err
+		}
+		network = models.NetworkFromString(networkStr)
+	}
+
+	if outputTxPath != "" {
+		if _, err := os.Stat(outputTxPath); err == nil {
+			return fmt.Errorf("outputTxPath %q already exists", outputTxPath)
+		}
+	}
+
+	if len(ledgerAddresses) > 0 {
+		useLedger = true
+	}
+
+	if useLedger && keyName != "" {
+		return ErrMutuallyExlusiveKeyLedger
+	}
+
+	switch network {
+	case models.Fuji:
+		if !useLedger && keyName == "" {
+			useLedger, keyName, err = prompts.GetFujiKeyOrLedger(app.Prompt, "pay transaction fees", app.GetKeyDir())
+			if err != nil {
+				return err
+			}
+		}
+	case models.Mainnet:
+		useLedger = true
+		if keyName != "" {
+			return ErrStoredKeyOnMainnet
+		}
+	default:
+		return errors.New("unsupported network")
+	}
+
+	// used in E2E to simulate public network execution paths on a local network
+	if os.Getenv(constants.SimulatePublicNetwork) != "" {
+		network = models.Local
+	}
+
+	if nodeIDStr == "" {
+		nodeID, err = subnetcmd.PromptNodeID()
+		if err != nil {
+			return err
+		}
+	} else {
+		nodeID, err = ids.NodeIDFromString(nodeIDStr)
+		if err != nil {
+			return err
+		}
+	}
+
+	minValStake, err := nodecmd.GetMinStakingAmount(network)
+	if err != nil {
+		return err
+	}
+	if weight == 0 {
+		weight, err = nodecmd.PromptWeightPrimaryNetwork(network)
+		if err != nil {
+			return err
+		}
+	}
+	if weight < minValStake {
+		return fmt.Errorf("illegal weight, must be greater than or equal to %d: %d", minValStake, weight)
+	}
+
+	kc, err := subnetcmd.GetKeychain(useLedger, ledgerAddresses, keyName, network)
+	if err != nil {
+		return err
+	}
+	jsonPop, err := promptProofOfPossession()
+	if err != nil {
+		return err
+	}
+	popBytes, err := json.Marshal(jsonPop)
+	if err != nil {
+		return err
+	}
+	start, duration, err = nodecmd.GetTimeParametersPrimaryNetwork(network, 0)
+	if err != nil {
+		return err
+	}
+	deployer := subnet.NewPublicDeployer(app, useLedger, kc, network)
+	nodecmd.PrintNodeJoinPrimaryNetworkOutput(nodeID, weight, network, start)
+	recipientAddr := kc.Addresses().List()[0]
```

Comment:
do we want to notify on where the rewards are going? do we want to ask the user for this?